cli/pcluster/config/config_patch.py

```diff
@@ -0,0 +1,297 @@
+                    )
+                )
+
+    def _adapt_from_target(self):
+        """Create a base section for each target section that is still not matched."""
+        for section_key in self.target_config.get_section_keys():
+            for _, target_section in self.target_config.get_sections(section_key).items():
+                base_section = self._get_base_section(target_section)
+                LOGGER.debug(
+                    "Matched target section [{key} {target_label}] with target section [{key} {base_label}]".format(
+                        key=section_key, base_label=base_section.label, target_label=target_section.label
+                    )
+                )
+
+    def _compare(self):
+        """
+        Compare the target config to the source one.
+
+        All sections in both configurations are compared one by one. This is made possible by the adaptation phase.
+        The last compared section is "cluster". This is compared only after the mock sections created during the
+        adaptation phase are removed, in order to make differences in settings parameters (like ebs_settings) appear
+        correctly.
+        """
+        # First, compare all sections except "cluster".
+        for section_key in self.target_config.get_section_keys():
+            if section_key != "cluster":
+                for _, target_section in self.target_config.get_sections(section_key).items():
+                    base_section = self.base_config.get_section(target_section.key, target_section.label)
+                    self._compare_section(base_section, target_section)
+
+        # Then, remove all mock sections. Settings_parameters will be refresh after this.
+        self._remove_mock_sections(self.base_config)
+        self._remove_mock_sections(self.target_config)
+
+        # Finally, compare cluster sections.
+        self._compare_section(
+            self.base_config.get_section("cluster"), self.target_config.get_section("cluster"),
+        )
+
+    def _compare_section(self, base_section, target_section):
+        for _, param in target_section.params.items():
+            base_value = base_section.get_param_value(param.key)
+            target_value = param.value
+
+            if base_value != target_value:
+                self.changes.append(
+                    Change(
+                        target_section.key,
+                        target_section.label,
+                        param.key,
+                        base_value,
+                        param.value,
+                        param.get_update_policy(),
+                    )
+                )
+
+    def _remove_ignored_sections(self, config):
+        for section_key in ConfigPatch.IGNORED_SECTIONS:
+            config.remove_section(section_key)
+
+    def _remove_mock_sections(self, config):
+        for section_key in config.get_section_keys():
+            for _, section in sorted(config.get_sections(section_key).items()):
```

Comment:
Nope, here is not necessary.

Reply:
However, we need to make a copy of the keys to avoid errors a runtime of type: "dictionary changed size during iteration". So I replaced sorted with "list".